linode_api4/objects/linode.py

```diff
@@ -765,20 +790,63 @@ def kvmify(self):
 
         return True
 
-    def mutate(self):
+    def mutate(self, allow_auto_disk_resize=True):
         """
         Upgrades this Instance to the latest generation type
         """
-        self._client.post('{}/mutate'.format(Instance.api_endpoint), model=self)
+
+        params = {
+            "allow_auto_disk_resize": allow_auto_disk_resize
+        }
+
+        self._client.post('{}/mutate'.format(Instance.api_endpoint), model=self, data=params)
 
         return True
 
-    def initiate_migration(self):
+    def initiate_migration(self, region=None, upgrade=None):
         """
         Initiates a pending migration that is already scheduled for this Linode
         Instance
         """
-        self._client.post('{}/migrate'.format(Instance.api_endpoint), model=self)
+        params = {
+            "region": region.id if issubclass(type(region), Base) else region,
+            "upgrade": upgrade
+        }
+
+        util.drop_null_keys(params)
+
+        self._client.post('{}/migrate'.format(Instance.api_endpoint), model=self, data=params)
+
+    def firewalls(self):
+        """
+        View Firewall information for Firewalls associated with this Linode.
+        """
+        from linode_api4.objects import Firewall # pylint: disable=import-outside-toplevel
```

Comment:
Just curious, why don't we have the import at the top of the file? It seems more common to have all the imports at the top level.

Reply:
Importing these at the top of the file caused a circular import error, so I had to import within the scope of the function.